### DISCUSSION TITLE
Trying to find a use case for the Glow layers of `resteasy-spring`

### DISCUSSION THREAD

**Discussion by liweinan** [resteasy/resteasy-spring#184] [Ideas]
@jfdenise I'm trying to write an example showing the usage of the Glow layers of the `resteasy-spring`, and I'm trying to add the layer configuration inside the `wildfly-maven-plugin` of the sample project. The problem is: it seems I don't need to add the layers configuration at all, because the provisioned WildFly server already shipped the `resteasy-spring` module(and hence no need to add a `feature-pack` section either) by default.

Do you have any suggestion on this topic? Or there are any modules/layers that is more properly to use as a use case of Glow?

**Comment by jfdenise** [comment]
You mean that WildFly natively supports resteasy-spring? From which JBoss Modules module the resteasy-spring module is referenced? We could ask @jamezp  to know what is the use-case of the resteasy-sping feature-pack? Replace the module version? Should resteasy-sping be removed from WildFly?

**Reply by jamezp** [reply]
Yes, WildFly provides a `org.jboss.resteasy.resteasy-spring` module and we cannot remove it from WildFly. We could make it a layer, but it's currently not a layer and just brought in via the `wildfly-ee-galleon-pack`.

**Reply by jfdenise** [reply]
So if we use layer, it shouldn't be brought in?

**Reply by jamezp** [reply]
What I think the issue is he was testing removing something that would trigger the layer coming in so Glow would think it's not required. The problem is WildFly always brings in the module if RESTEasy is brought in, or some other trigger I haven't looked :). 

The "fix" would be to add a layer in WildFly for the `org.jboss.resteasy.resteasy-spring` module that would match what we have in resteasy-spring.

This is likely an RFE for WildFly, but IMO it's reasonable and likely good to have. There is no real reason to include the `org.jboss.resteasy.resteasy-spring` module for most deployments.

**Reply by liweinan** [reply]
@jamezp Thanks for explaining the requirement! :D  @jfdenise As James said, currently it seems I can't find a use case to use Glow layers configuration in the `wildfly-maven-plugin`. The problem is:

- The `resteasy-spring` is already bundled inside provisioned WildFly server, so I don't need to add Glow layers configuration(and it doesn't need Glow to detect it automatically, because it's already bundled).
- If I'm trying to find a module that is not bundled inside WildFly, and it needs Glow to detect it, then firstly, this module should have a `feature-pack` in WildFly in advance. However, in such case, this module is already bundled inside WildFly(Or maybe I haven't found a module that meets this siutation).

I'm not sure whether my understanding is correct or not. In conclusion, I want to provide a use case in a sample project that uses `wildfly-maven-plugin` and also Glow features(for example, put the layers configuration in the plugin, or let the layers to be automatically dectected).

I'm still learning on using Glow and maybe my understanding is not correct, please correct if I'm wrong on some points :D

**Reply by jfdenise** [reply]
What I would like to understand is how the resteasy-spring is already bundled. JBoss Modules are provisioned only if they are referenced from the extension (transitively). If one provision the simplest server (image undertow web server), the spring-resteasy module is also provisioned?

**Reply by liweinan** [reply]
@jfdenise Here is the quickstart example that shows the use case of `wildfly-maven-plugin` with `spring-resteasy`: 

- https://github.com/wildfly/quickstart/tree/main/spring-resteasy

Which doesn't involve Glow or `feature-pack` configurations. What I want is to provide an example that explicitly needs Glow layer detection feature.

**Reply by jamezp** [reply]
No layers are defined in that example so it will configure the whole server. What I believe @jfdenise is suggesting is that you add a simple layer like undertow which would not bring in the `jaxrs` layer. Then Glow should determine that the `resteasy-spring` layer is required. That said you'd have to ensure you add the `resteasy-spring-galleon-pack` feature pack as well.

That will work for testing, but we can't have it in the wildfly quickstarts.

**Reply by liweinan** [reply]
@jamezp Cool! I'll try to write one sample like this and let you know about my findings. @jfdenise Thanks for help too!
